src/cashlib/Test_libbitcoincash_android/app/src/androidTest/java/info/bitcoinunlimited/www/test_libbitcoincash_android/InstrumentedTest.kt

```diff
@@ -0,0 +1,138 @@
+package info.bitcoinunlimited.www.test_libbitcoincash_android
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import bitcoinunlimited.libbitcoincash.*
+
+import org.junit.Test
+import org.junit.runner.RunWith
+
+import org.junit.Assert.*
+import java.lang.AssertionError
+import java.math.BigInteger
+import java.util.logging.Logger
+
+val LogIt = Logger.getLogger("AndroidTest")
+
+// Troubleshooting:
+
+// ---
+// native compile: gmp.h not found
+
+// you have a generated file "src/secp256k1/src/libsecp256k1-config.h".  This was generated for your native build.  You need to use the one in src/cashlib.  To do so, just delete
+// "src/secp256k1/src/libsecp256k1-config.h".  Tip: if you do out-of-source native builds you will be able to have both android and normal builds simultaneously.
```

Comment:
It needs to be the same file name because that's the header name included in all the files.  We do have one for a normal build and one for android because android is a crossbuild and this file basically lets you select architecture-specific optimizations.  My android version selects no optimizations.  Someday we could make this better, but we don't really need optimized signature generation or checking on Android right now.

Reply:
The android cashlib test was removed from this PR since it was agreed that its just too much extraneous "stuff" to add to the repo right now.  The functionality will be tested in another repo.